SPECS/grub2/grub2.spec

```diff
@@ -269,9 +269,31 @@ GRUB_PXE_MODULE_SOURCE=%{buildroot}%{_datadir}/grub2-efi/grubaa64-noprefix.efi
 cp $GRUB_MODULE_SOURCE $EFI_BOOT_DIR/$GRUB_MODULE_NAME
 cp $GRUB_PXE_MODULE_SOURCE $EFI_BOOT_DIR/$GRUB_PXE_MODULE_NAME
 
-%post   -p /sbin/ldconfig
+%post
+/sbin/ldconfig
+# When grub2 is installed, we need to populate the defaults file, assuming that a user
+# may invoke grub tools such as grub2-mkconfig for grub updates, in its current state, running
+# grub2-mkconfig on Mariner will brick Mariner, as there is an assumption that the grub.cfg file
+# is setup and configured only once during bootstrap of the image.
+
+# Only update the file if the file exists and is empty, take last known good CMDLINE configuration
```

Comment:
The way to do this is to list the file in the normal %files section, but as a config file.
https://rpm-software-management.github.io/rpm/manual/spec.html#config